lib/crypto/ed25519/ed25519_test.go

```diff
@@ -99,3 +102,54 @@ func TestNewKeypairFromMnenomic_Again(t *testing.T) {
 	expectedPubkey := common.MustHexToBytes("0xf56d9231e7b7badd3f1e10ad15ef8aa08b70839723d0a2d10d7329f0ea2b8c61")
 	require.Equal(t, expectedPubkey, kp.Public().Encode())
 }
+
+func TestVerifySignature(t *testing.T) {
+	keypair, err := GenerateKeypair()
+	require.NoError(t, err)
+
+	message := []byte("Hello world!")
+
+	signature, err := keypair.Sign(message)
+	require.NoError(t, err)
+
+	testCase := map[string]struct {
+		publicKey, signature, message []byte
+		err                           error
+	}{
+		"success": {
+			publicKey: keypair.public.Encode(),
+			signature: signature,
+			message:   message,
+			err:       nil,
+		},
+		"failed to fetch publicKey": {
+			publicKey: []byte(""),
+			signature: signature,
+			message:   message,
+			err:       errors.New("cannot create public key: input is not 32 bytes"),
+		},
+		"verification failed": {
+			publicKey: keypair.public.Encode(),
+			signature: []byte(""),
```

Comment:
Supernit no need for conversion string to bytes

```suggestion
			signature: []byte{},
```